File: Mobile/SchedHoliday/app/src/main/java/be/helmo/schedholiday/repository/User/DTOUser.kt
package be.helmo.schedholiday.repository.User

import kotlinx.serialization.Serializable
import kotlinx.serialization.encodeToString
import kotlinx.serialization.json.Json
@Serializable
open class DTOUser(
    val Id:String,
    val FirstName:String,
    val LastName:String,
    val Email:String,
)   {
    fun toJson() : String{
        return Json.encodeToString(this)
    }
}

File: API/SchedHoliday/Repo/User/DTOUser.cs
using Microsoft.AspNetCore.Identity;
using SchedHoliday.Repo.Holiday;
using SchedHoliday.Repo.Message;
using System.ComponentModel.DataAnnotations.Schema;

namespace SchedHoliday.Repo.User
{
    public class DTOUser : IdentityUser
    {

        public string Id { get; set; }

        public string FirstName { get; set; }

        public string LastName { get; set; }

        public string Email { get; set; }

        public bool fromOAuth { get; set ; } 

        [NotMapped]
        public string Password { get; set; }

        public IEnumerable<DTOMessage>? Messages { get; set; }

        public IEnumerable<DTOHoliday>? Holidays { get; set; }

        public override string ToString()
        {
            return Id.ToString();
        }
    }
}
